tiny_dnn/layers/recurrent_cell_layer.h

```diff
@@ -65,9 +65,22 @@ class recurrent_cell_layer : public layer {
                        bool has_bias                = true,
                        activation_layer *activation = new tanh_layer,
                        core::backend_t backend_type = core::default_engine())
-    : layer(recurrent_order(has_bias),  // output bias
-            {vector_type::data,         // output vector
-             vector_type::aux}) {
+    : layer({vector_type::data, vector_type::data, vector_type::data,
+             vector_type::data, vector_type::data},
+            {vector_type::data, vector_type::data}) {
+    layer::add_parameter(1, 1, 1, in_dim, parameter_type::weight, true);  // x
+    layer::add_parameter(1, 1, 1, out_dim, parameter_type::weight,
+                         true);  // h(t-1)
+    layer::add_parameter(1, 1, out_dim, in_dim, parameter_type::weight,
+                         true);  // U
+    layer::add_parameter(1, 1, out_dim, out_dim, parameter_type::weight,
+                         true);  // W
+    layer::add_parameter(1, 1, out_dim, out_dim, parameter_type::weight,
+                         true);  // V
+    if (has_bias) {
+      layer::add_parameter(1, 1, 1, out_dim, parameter_type::bias, true);  // b
+      layer::add_parameter(1, 1, 1, out_dim, parameter_type::bias, true);  // c
+    }
```

Comment:
@karandesai-96 missing something since I'm getting this error:
`C++ exception with description "Connection mismatch at setup layer" thrown in the test body.`

/cc @prlz77